### DISCUSSION TITLE
Default component for Date fieldtypes

### DISCUSSION THREAD

**Discussion by MDrooker** [vazco/uniforms#1009] [Q&A] [answered]
Noob question-
I have a schema that has dates in it.  
Some of the records being edited still have null/empty for the date fields(there are a few).

I created my own DateField that handles nulls and string based dates(like (2021-08-29T02:12:11.849Z).  
And just converts it to 1/1/1980 or a valid date object
And know I can-
``` <AutoField name={'startDateTime'} component={DateTimeField} />```  to handle those fields one at a time.

But there are a bunch of Date fields(and some are nested).

Question-
**Is it possible to tell the ```<AutoForm>``` that I want to always use my ```<DateTimeField>``` when a date field is used in the schema?**  
That way I dont have to cherry pick each of the date fields to display.

Because right now for the fields that are null.  I get -
```TypeError: value.toISOString is not a function```

**Comment by wadamek65** [comment]
Hi @MDrooker 

Yes it is possible to tell `AutoForm` to use your own custom component instead. You should be able to solve your problem by creating your own `AutoField` and overriding its behavior regarding `Date` types. Then you want to pass your custom `AutoField` to the `AutoForm`.

You can find an example in the documentation here: https://uniforms.tools/docs/examples-custom-fields#customautofield 

**Small note**: I noticed that the link to the original code in the aforementioned docs isn't working, so here is the working one: https://github.com/vazco/uniforms/blob/master/packages/uniforms-unstyled/src/AutoField.tsx

**Reply by MDrooker** [reply]
Argh...That make total sense.  And is really straightforward.  Thanks!

**Reply by wadamek65** [reply]
Glad I could help. If you have any other questions feel free to create a new issue/discussion and we'll do our best to help.

**Reply by MDrooker** [reply]
@wadamek65 One small note...I also had to override the ```<NestField />``` component to flow downward to nested components the new ```<AutoField />``` component.  It was using the default ```<AutoField />``` for nested form elements.

**Reply by wadamek65** [reply]
@MDrooker Hmm, thank you very much for this insight, I haven't taken this into consideration. Also, having to do this sounds rather unintuitive - I'll see if maybe we would like to change this behavior.

**Reply by MDrooker** [reply]
Glad to give new insight.

Forgot to add...I did update ```<ListItemField />``` too with the new reference to ```<AutoField />```

**Comment by radekmie** [comment, marked as answer]
_Technically_, the `autoForm` that @wadamek65 mentioned is no longer the preferred option, due to the exact problems you've faced. You should use the [`AutoField.componentDetectorContext`](https://uniforms.tools/docs/uth-autofield-algorithm#overriding-autofield) instead - then all of the components are correctly resolved, even when deeply nested.

I'll update the documentation accordingly.

**EDIT:** See #980.
